src/parser/FStar.Parser.Dep.fsi

```diff
@@ -27,15 +27,22 @@ val hard_coded_dependencies : string -> list<(lident * open_kind)>
 
 val is_interface: string -> bool
 val is_implementation: string -> bool
-
+type module_name = string
+type dependence =
+    | UseInterface of module_name
+    | PreferInterface of module_name
+    | UseImplementation of module_name
+    | FriendImplementation of module_name
+type parsing_data = list<dependence> * list<dependence> * bool
```

Comment:
can this be abstract too? 

For F#, you will need to make parsing_data a new inductive type, rather than just a type abbreviations.
You will also probably need to provide a empty_parsing_data value (for use in FStar.Interactive.Legacy)

Reply:
Thanks for noting it, should be possible, I will try.

Reply:
Fixed in https://github.com/FStarLang/FStar/commit/f5ba9096b86f8a0cf4243c587cb2ef8f1ccbb140#diff-41fa99e53b70e2e0f9a569c692749585.